src/Services/Azure/DataExplorer/DataExplorerService.cs

```diff
@@ -0,0 +1,315 @@
+        while (reader.Read())
+        {
+            var jsonString = reader["Schema"].ToString()!;
+            var json = JsonDocument.Parse(jsonString);
+            result.Add(json);
+            
+        }
+        return result;
+    }
+
+    public async Task<List<JsonDocument>> QueryItems(
+        string subscriptionId, 
+        string clusterName, 
+        string databaseName, 
+        string query, 
+        string? tenant = null, 
+        AuthMethod? authMethod = AuthMethod.Credential, 
+        RetryPolicyArguments? retryPolicy = null)
+    {
+        ValidateRequiredParameters(subscriptionId, clusterName, databaseName, query);
+
+        string clusterUri = await GetClusterUri(subscriptionId, clusterName, tenant, retryPolicy);
+
+        return await QueryItems(clusterUri, databaseName, query, tenant, authMethod, retryPolicy);
+    }
+
+    private async Task<string> GetClusterUri(string subscriptionId, string clusterName, string? tenant, RetryPolicyArguments? retryPolicy)
+    {
+        var cluster = await GetCluster(subscriptionId, clusterName, tenant, retryPolicy) ?? throw new Exception($"Data Explorer cluster '{clusterName}' not found in subscription '{subscriptionId}'.");
+
+        if (!cluster.RootElement.TryGetProperty("ClusterUri", out var clusterUriElement))
+        {
+            throw new Exception($"Could not retrieve URI for cluster '{clusterName}'");
+        }
+
+        var clusterUri = clusterUriElement.GetString();
+        if (string.IsNullOrEmpty(clusterUri))
+        {
+            throw new Exception($"Could not retrieve URI for cluster '{clusterName}'");
+        }
+
+        return clusterUri!;
+    }
+
+    public async Task<List<JsonDocument>> QueryItems(
+        string clusterUri, 
+        string databaseName, 
+        string query, string? tenant = null, 
+        AuthMethod? authMethod = AuthMethod.Credential, 
+        RetryPolicyArguments? retryPolicy = null)
+    {
+        ValidateRequiredParameters(clusterUri, databaseName, query);
+
+        var kcsb = await CreateKustoConnectionStringBuilder(
+            clusterUri,
+            authMethod,
+            null,
+            tenant);
+            
+        using var queryProvider = KustoClientFactory.CreateCslQueryProvider(kcsb);
+        var reader = await queryProvider.ExecuteQueryAsync(databaseName, query, null);
+        var results = new List<JsonDocument>();
+
+        try
+        {
+            while (reader.Read())
+            {
+                var dict = new Dictionary<string, object?>();
+                for (int i = 0; i < reader.FieldCount; i++)
+                {
+                    dict[reader.GetName(i)] = reader.GetValue(i);
+                }
+                var json = JsonSerializer.SerializeToDocument(dict);
+                results.Add(json);
+            }
+        }
+        catch (Exception ex)
+        {
+            throw new Exception($"Error executing query: {ex.Message}", ex);
```

Comment:
Do you see added value in wrapping in a generic exception? Kusto exceptions are pretty descriptive as they are